File: src/test/java/organizing/data/extractConstants.java
package organizing.data;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class extractConstants {
    @Test
    public void validatePassword() {
        String validPassword = "OJ°0I°09121l.";
        assertDoesNotThrow(() -> Utils.validatePassword(validPassword));
    }

    @Test
    public void invalidatePassword() {
        String invalidPassword = "Im invalid";
        assertThrows(IllegalArgumentException.class, () -> Utils.validatePassword(invalidPassword));
    }

    @Test
    public void calculatePotentialEnergy() {
        assertEquals(9.81, Utils.calculatePotentialEnergy(1, 1));
        assertEquals(156332.7, Utils.calculatePotentialEnergy(98.9897, 160.987), 0.01);
        assertEquals(0, Utils.calculatePotentialEnergy(0, 1));
        assertEquals(0, Utils.calculatePotentialEnergy(1, 0));
    }
}

File: src/scala/src/test/scala/organizing/data/extractConstants.scala
package org.ythirion.refactoring.journey
package organizing.data

import organizing.data.Utils.{calculatePotentialEnergy, validatePassword}

import org.scalactic.Tolerance.convertNumericToPlusOrMinusWrapper
import org.scalatest.TryValues.convertTryToSuccessOrFailure
import org.scalatest.funsuite.AnyFunSuite

class extractConstants extends AnyFunSuite {

  test("validPassword returns a success for password following pwd rules") {
    val validPassword = "OJ°0I°09121l."
    assert(validatePassword(validPassword).success.value)
  }

  test("validPassword returns a failure for password not following pwd rules") {
    val invalidPassword = "Im invalid"
    assert(validatePassword(invalidPassword).failure.exception.getMessage == "minimum password length is not respected")
  }

  test("calculate potential energy") {
    assert(calculatePotentialEnergy(1, 1) == 9.81)
    assert(calculatePotentialEnergy(98.9897, 160.987) === 156332.7 +- 0.01)
    assert(calculatePotentialEnergy(0, 1) == 0)
    assert(calculatePotentialEnergy(1, 0) == 0)
  }
}